File: ai-workspace-web/src/api/monitor.ts
/**
 * 系统监控 API（仅管理员）
 *
 * 对应后端 workspace-monitor 模块的 MonitorController，无 DB 表。
 * ServerInfo 数据来自 JDK MXBeans；ServiceHealth 通过探测 Redis/FastAPI/MinIO 获得。
 */
import request from './request'

/** 服务器运行时指标，由 JDK MXBeans 采集 */
export interface ServerInfo {
  cpu: { cores: number; usedPercent: number; procUsedPercent: number }
  memory: { total: number; used: number; usedPercent: number }
  runtime: {
    version: string
    numGoroutine: number
    heapAlloc: number
    heapSys: number
    heapInuse: number
    usedPercent: number
  }
  os: { goos: string; goarch: string }
  disks: { path: string; total: number; used: number; usedPercent: number }[]
}

/** 单个依赖服务的健康检查结果 */
export interface ServiceHealth {
  name: string
  status: 'UP' | 'DOWN'
  /** 被探测的 URL 或地址 */
  target: string
  /** 探测延迟（毫秒），DOWN 时无意义 */
  latency: number
  message: string | null
}

export const getServerInfo = () =>
  request.get<unknown, ServerInfo>('/monitor/server')

export const getServiceHealth = () =>
  request.get<unknown, ServiceHealth[]>('/monitor/health')

File: ai-workspace-backend/internal/handler/monitor.go
package handler

import (
	"github.com/gin-gonic/gin"

	"github.com/aiworkspace/backend/internal/common"
	"github.com/aiworkspace/backend/internal/service"
)

// MonitorServer GET /api/monitor/server — 仅管理员，采集服务器运行时指标快照
//
// 指标通过 gopsutil 跨平台获取，包含 CPU 使用率、物理内存、Go 运行时（等价 JVM 堆）、
// OS 信息与各磁盘分区使用情况，按请求实时采样，不持久化。
func MonitorServer(c *gin.Context) {
	common.OK(c, service.MonitorSvc.GetServerInfo())
}

// MonitorHealth GET /api/monitor/health — 仅管理员，探测依赖服务健康状态
//
// 探测 Redis（PING）、FastAPI（GET /health）、MinIO（GET /minio/health/live），
// 每个探针超时 2 秒，结果含 status(UP/DOWN)、延迟(ms)和错误信息。
func MonitorHealth(c *gin.Context) {
	common.OK(c, service.MonitorSvc.GetServiceHealth())
}
